interfaces/builtin/ssh-control.go

```diff
@@ -0,0 +1,39 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const sshControlConnectedPlugAppArmor = `
+# Description: Allow modifying the system SSH configuration
+
+/etc/ssh/sshd_not_to_be_run rw,
+`
+
+// NewSshControlInterface returns a new "ssh-control" interface.
+func NewSshControlInterface() interfaces.Interface {
+	return &commonInterface{
+		name: "ssh-control",
```

Comment:
@jdstrand Your proposal sounds pretty good to me. I can go and change this to implement the sshd-control part.

@ogra1 @jdstrand I am fine if we limit sshd-control for now to be only used from the core (configure hook) and maybe also from the gadget. In the configure hook we can actually verify if there is a user available with an active password set to ensure at least an emergency option for users to recover the device when a serial or monitor port is available. However on headless devices this doesn't help at all so I think the only option is that we grant this only through the configure hook which can do some extra validation and can be only called by a logged in user or by a snap having access to the privileged snapd-control interface. What do you think? 

@niemeyer @mvo5 Can you have a look at this too?